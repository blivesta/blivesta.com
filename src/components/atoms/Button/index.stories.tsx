import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Button from '.';

const Wrapper = styled.div`
  padding: 60px;
`;

storiesOf(`atoms|Button`, module).add(`default`, () => (
  <Wrapper>
    <Button type="submit">Button</Button>
  </Wrapper>
));
