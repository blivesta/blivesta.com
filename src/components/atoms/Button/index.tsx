import * as React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  outline: none;
  appearance: none;
  background-image: none;
  text-decoration: none;
  user-select: none;
  white-space: nowrap;
  background-color: transparent;
  color: ${props => props.theme.colors.baseText[0]};
  font-weight: normal;
  font-family: ${props => props.theme.fonts.avantGarde};
  font-size: ${props => props.theme.fontSizes[1]};
  padding: 0 ${props => props.theme.spaces[4]};
  height: 40px;
  border: 2px solid ${props => props.theme.colors.baseText[0]};
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  text-transform: uppercase;
  text-decoration: none;
  letter-spacing: 0.5em;
  text-indent: 0.5em;
  transition: background-color, color 0.2s;

  &:last-child {
    margin-right: 0;
  }
  &:hover,
  &:focus,
  &:active {
    background-color: ${props => props.theme.colors.baseText[0]};
    color: ${props => props.theme.colors.bg};
  }
`;

export interface ButtonProps {
  children: React.ReactNode;
  type?: 'submit' | 'reset' | 'button';
  name?: string;
  disabled?: boolean;
  onClick?: (e: React.MouseEvent<HTMLElement>) => void;
}

const Button = ({ children, type, name, disabled, onClick }: ButtonProps) => (
  <StyledButton type={type} name={name} disabled={disabled} onClick={onClick}>
    {children}
  </StyledButton>
);

export default Button;
