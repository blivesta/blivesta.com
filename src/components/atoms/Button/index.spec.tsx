import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '.';

describe('Button', () => {
  it('snapshot', () => {
    const tree = shallow(<Button>...</Button>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
