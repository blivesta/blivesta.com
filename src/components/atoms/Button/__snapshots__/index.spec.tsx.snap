// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<styled.button>
  ...
</styled.button>
`;
